app/src/main/java/com/github/se/cyrcle/model/report/ReportedObjectViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.github.se.cyrcle.model.report
+
+import androidx.lifecycle.ViewModel
+import com.github.se.cyrcle.model.parking.ParkingReport
+import com.github.se.cyrcle.model.review.ReviewReport
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+sealed class Report {
+  data class Parking(val parkingReport: ParkingReport) : Report()
+
+  data class Review(val reviewReport: ReviewReport) : Report()
+}
+
+class ReportedObjectViewModel(private val reportedObjectRepository: ReportedObjectRepository) :
+    ViewModel() {
+
+  private val _selectedObject = MutableStateFlow<ReportedObject?>(null)
+  val selectedObject: StateFlow<ReportedObject?> = _selectedObject
+
+  private val _reportsList = MutableStateFlow<List<ReportedObject>>(emptyList())
+  val reportsList: StateFlow<List<ReportedObject>> = _reportsList
+
+  fun getNewUid(): String {
+    return reportedObjectRepository.getNewUid()
+  }
+
+  fun deleteReportedObject(uid: String) {
+    reportedObjectRepository.deleteReportedObject(uid, {}, {})
+  }
+
+  fun selectObject(reportedObject: ReportedObject) {
+    _selectedObject.value = reportedObject
+  }
+
+  fun clearSelectedObject() {
```

Comment:
Same here, this function is never used. Maybe it's for later then it's fine i guess...

Reply:
Yep, I'll need it soon!